File: pinyougou-parent/pinyougou-shop-web/src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../address/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../address/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findById=function(id){
		return $http.get('../address/findById.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../address/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../address/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../address/delete.php?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../address/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/AddressService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbAddress;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface AddressService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbAddress> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findByPagination(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(TbAddress address);


    /**
     * 修改
     */
    void update(TbAddress address);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbAddress findById(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum
     *         当前页 码
     * @param pageSize
     *         每页记录数
     * @return
     */
    PageResult findWithConditionsByPagination(TbAddress address, int pageNum, int pageSize);

}
